x/ibc/core/02-client/types/codec.go

```diff
@@ -11,6 +12,15 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/ibc/core/exported"
 )
 
+var (
+	// SubModuleCdc references the global 02-client sub module codec. Note, the codec
+	// should ONLY be used in certain instances of tests and for JSON encoding.
+	//
+	// The actual codec used for serialization should be provided to 02-client and
+	// defined at the application level.
+	SubModuleCdc = codec.NewProtoCodec(codectypes.NewInterfaceRegistry())
```

Comment:
q: don't you need to register the interfaces first?

Reply:
I'm not sure what I was thinking. This is being done in the keeper. I can use the keeper's cdc

Reply:
ah, the cdc is only a binary marshaler. I'm switching to proto so removing this. I think you are correct though